pylint/checkers/variables.py

```diff
@@ -358,12 +358,13 @@ def _assigned_locally(name_node):
     return any(a.name == name_node.name for a in assign_stmts)
 
 
-def _is_type_checking_import(node):
-    parent = node.parent
-    if not isinstance(parent, nodes.If):
-        return False
-    test = parent.test
-    return test.as_string() in TYPING_TYPE_CHECKS_GUARDS
+def _is_type_checking_import(node: Union[nodes.Import, nodes.ImportFrom]) -> bool:
+    """Check if an import node is guarded by a TYPE_CHECKS guard"""
+    for ancestor in node.node_ancestors():
+        if isinstance(ancestor, nodes.If):
+            if ancestor.test.as_string() in TYPING_TYPE_CHECKS_GUARDS:
```

Comment:
I'm again late, as I'm going through my notifications.
Can't those two be combined? Technically even a `return any(...)` would be possible, but I think that might be more difficult to read.

Reply:
Yeah, should have catched that myself. Going to make a PR with this change!

`consider-combining-if` seems like a handy checker for our review process 😄

Reply:
> `consider-combining-if` seems like a handy checker for our review process 😄

That is also somewhere on my ideas list 😅

Reply:
I found the `any` to be okay for readability. See #5410

Reply:
> consider-combining-if seems like a handy checker for our review process

Why didn't I think of that sooner, It's like half the content of my reviews 😄 Also it should trigger on thing like  ``any(bool(e == 1) for e in elements if int(e) == e))``. Let me open an issue.

Reply:
https://github.com/PyCQA/pylint/issues/5414